packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2601,17 +2610,56 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         serializedContent: string,
         serverMaxOpSize: number,
         batch: boolean,
-        opMetadataInternal: unknown = undefined,
+        opMetadataInternal: Record<string, unknown> | undefined = undefined,
     ): number {
+        let maybeCompressedContent: any;
+        let metadata: Record<string, unknown> | undefined;
+        let contentLength: number;
+
+        if (this.runtimeOptions.compressionOptions?.minimumSize !== undefined &&
+            serializedContent &&
+            serializedContent.length > this.runtimeOptions.compressionOptions.minimumSize
+        ) {
+            this.compressedOpCount++;
+
+            const compressionStart = Date.now();
+            const contentsAsBuffer = new TextEncoder().encode(serializedContent);
```

Comment:
nitpick: consider making TextEncoder() and TextDecoder() private member variables to avoid repeatedly creating a new instance.